Dockerfile.alpine

```diff
@@ -13,13 +13,17 @@
 # limitations under the License.
 
 # Use the latest stable golang 1.x to compile to a binary
-FROM golang:1 as build
+FROM --platform=$BUILDPLATFORM golang:1 as build
 
 WORKDIR /go/src/cloudsql-proxy
 COPY . .
 
+ARG TARGETOS
+ARG TARGETARCH
+
 RUN go get ./...
-RUN go build -ldflags "-X main.metadataString=container.alpine" -o cloud_sql_proxy ./cmd/cloud_sql_proxy
+RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} \
```

Comment:
My uninformed opinion is that we should always build static binaries, including for Alpine and Buster. However, this is a change from previous versions. Do we need to enable this for the multi arch builds?

Reply:
Tested this removing the `CGO_ENABLED=0` flag on my M1 mac and it works, so I agree, let's remove from the diff